File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log32008/diet.cpp
/*
TASK:diet
LANG:C++
*/
#include <iostream>
using namespace std;

int main()
{
    int a,b,c;
    int days=0;
    cin >> a >> b >> c;
    days=(a*10+b*20+c*30)/60;
    
    cout << days << " ";
    
    long long br=0;
    for(;days;days--)
    {
                     if(a>=1 && b>=1 && c>=1) { a--; b--; c--; continue; }
                     //cout << a << " " << b << " " << c << endl;
                     if(a==0)
                     {
                             if(b<=c) { br++; a=1; b++; c--; }
                             else { br++; a=2; b--; }
                     }
                     if(b==0)
                     {
                             if(a>2) { a-=2; b=1; }
                             else { br++; a++; b=1; c--; }
                     }
                     if(c==0)
                     {
                             if(a>1 && b>1) { a--; b--; c=1; }
                             else if(a>1 && b<=1) { a-=3; c=1; }
                             else if(a<=1 && b>1) { br++; a++; b-=2; c=1; }
                     }
                     //cout << a << " " << b << " " << c << endl;
                     a--; b--; c--;
                     //cout << a << " " << b << " " << c << endl;
                     //cout << endl;
    }
    cout << br << endl;
    
    return 0;
}

File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log32016/diet.pas
{
TASK:DIET
LANG:PASCAL
}

program diet;
var one,two,three,gr,days,cut:longint;
begin
{  assign(input,'diet.in'); reset(input);
  assign(output,'diet.out'); rewrite(output);}
  readln(one,two,three);
  gr:=one+two*2+three*3;
  days:=gr div 6;
  one:=one-days;
  two:=two-days;
  three:=three-days;
  while three<0 do
    begin
      one:=one-1;
      two:=two-1;
      three:=three+1;
    end;
  while two<0 do
    begin
      while (two<0) and (one>1) do
        begin
          one:=one-2;
          two:=two+1;
        end;
      while (two<0) and (three>0) do
        begin
          one:=one+1;
          cut:=cut+1;
          two:=two+1;
          three:=three-1;
          while (two<0) and (one>1) do
            begin
              one:=one-2;
              two:=two+1;
            end;
        end;
      if two<0 then
        begin
          one:=one-2;
          two:=two+1;
        end;
    end;
  while one<0 do
    begin
      while (one<0) and (two>0) do
        begin
          cut:=cut+1;
          one:=one+2;
          two:=two-1;
        end;
      if one<0 then
        begin
          one:=one+1;
          cut:=cut+1;
          two:=two+1;
          three:=three-1;
        end;
    end;
  while two<0 do
    begin
      cut:=cut+1;
      three:=three-1;
      one:=one+1;
      two:=two+1;
    end;
  writeln(days,' ',cut);
{  close(input);
  close(output); }
end.